### PR TITLE
Enable unit test and add loss unit tests

### PR DESCRIPTION
**Pull request by thomasdhc**

### CODE REVIEW

File: tests/unit_tests/loss/test_masked_ce.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pytest
+import torch
+import torch.nn.functional as F
+from nemo_automodel.loss.masked_ce import masked_cross_entropy
+
+
+def test_masked_cross_entropy_no_mask():
+    """
```
Comment (thomasdhc) [inline tests/unit_tests/loss/test_masked_ce.py:21]: @akoumpa I've added missing doc-strings here, please review and update as needed. Feel free to push directly to this branch

File: tests/unit_tests/loss/test_masked_ce.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    loss_ref = F.cross_entropy(logits, targets)
+
+    # They should be very close
+    assert torch.allclose(
+        loss_custom, loss_ref
+    ), f"Loss without mask expected {loss_ref.item():.4f}, but got {loss_custom.item():.4f}"
+
+
+def test_masked_cross_entropy_with_mask():
+    """
```
Comment (thomasdhc) [inline tests/unit_tests/loss/test_masked_ce.py:44]: @akoumpa doc strings review

File: tests/unit_tests/loss/test_masked_ce.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    loss_ref = F.cross_entropy(logits, targets_ref)
+
+    assert torch.allclose(
+        loss_custom, loss_ref
+    ), f"Loss with mask expected {loss_ref.item():.4f}, but got {loss_custom.item():.4f}"
+
+
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA not available")
+def test_masked_cross_entropy_gpu():
+    """
```
Comment (thomasdhc) [inline tests/unit_tests/loss/test_masked_ce.py:70]: @akoumpa doc strings review
